File: LeetCode/Isomorphic Strings.cpp
class Solution {
 public:
  bool isIsomorphic(string s, string t) {
    map<char, char> m, n;

    for (int i{}; i < s.size(); i++) {
      if (m.count(s[i]) && m[s[i]] != t[i] || n.count(t[i]) && n[t[i]] != s[i])
        return false;
      else {
        m[s[i]] = t[i];
        n[t[i]] = s[i];
      }
    }

    return true;
  }
};

File: LeetCode/Isomorphic Strings.ts
function isIsomorphic(s: string, t: string): boolean {
  const mapS_T:{[key: string]: string} = {}, mapT_S:{[key: string]: string} = {};

  for (let i = 0; i < s.length; i++) {
    if (
        mapS_T[s[i]] !== undefined && mapS_T[s[i]] !== t[i] 
        || mapT_S[t[i]] !== undefined && mapT_S[t[i]] !== s[i]
    )
      return false;
    else {
      mapS_T[s[i]] = t[i];
      mapT_S[t[i]] = s[i];
    }
  }

  return true;
};